agent.go

```diff
@@ -468,8 +468,12 @@ func (n *network) addDriverWatches() {
 
 	c := n.getController()
 	for _, tableName := range n.driverTables {
-		ch, cancel := c.agent.networkDB.Watch(tableName, n.ID(), "")
 		c.Lock()
+		if c.agent == nil {
```

Comment:
But I want to keep the current logic where the lock is held until after `c.agent.driverCancelFuncs` map is modified.

If you  are suggesting to add a local variable even thoug we will use it inside a locked scope, I do not see the need.